packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/create-model-customization-job.ts

```diff
@@ -0,0 +1,428 @@
+   *
+   * @default - no VPC configuration
+   */
+  readonly vpcConfig?: IBedrockCreateModelCustomizationJobVpcConfig;
+}
+
+/**
+ * A Step Functions Task to create model customization in Bedrock.
+ */
+export class BedrockCreateModelCustomizationJob extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+  private _role: iam.IRole;
+
+  constructor(scope: Construct, id: string, private readonly props: BedrockCreateModelCustomizationJobProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    this.validateStringLength('clientRequestToken', 1, 256, props.clientRequestToken);
+    this.validatePattern('clientRequestToken', /^[a-zA-Z0-9](-*[a-zA-Z0-9])*$/, props.clientRequestToken);
+    this.validateStringLength('customModelName', 1, 63, props.customModelName);
+    this.validatePattern('customModelName', /^([0-9a-zA-Z][_-]?)+$/, props.customModelName);
+    this.validateArrayLength('customModelTags', 0, 200, props.customModelTags);
+    this.validateStringLength('jobName', 1, 63, props.jobName);
+    this.validatePattern('jobName', /^[a-zA-Z0-9](-*[a-zA-Z0-9\+\-\.])*$/, props.jobName);
+    this.validateArrayLength('jobTags', 0, 200, props.jobTags);
+    this.validateArrayLength('validationDataS3Uri', 1, 10, props.validationDataS3Uri);
+    this.validateArrayLength('securityGroups', 1, 5, props.vpcConfig?.securityGroups);
+    this.validateArrayLength('subnets', 1, 16, props.vpcConfig?.subnets);
+
+    validatePatternSupported(this.integrationPattern, BedrockCreateModelCustomizationJob.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this._role = this.renderBedrockCreateModelCustomizationJobRole();
+    this.taskPolicies = this.renderPolicyStatements();
+  }
+
+  /**
+   * The IAM role for the bedrock create model customization job
+   */
+  public get role(): iam.IRole {
+    return this._role;
+  }
+
+  /**
+   * Configure the IAM role for the bedrock create model customization job
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/userguide/model-customization-code-samples.html
+   */
+  private renderBedrockCreateModelCustomizationJobRole(): iam.IRole {
+    if (this.props.role) {
+      return this.props.role;
+    }
+    const stack = Stack.of(this);
+    const role = new iam.Role(this, 'BedrockRole', {
+      assumedBy: new iam.ServicePrincipal('bedrock.amazonaws.com'),
+      inlinePolicies: {
+        BedrockCreateModelCustomizationJob: new iam.PolicyDocument({
+          statements: [
+            ...(this.props.vpcConfig ? [
```

Comment:
Can you please create a separate private method for returning the VPC-related statements?